public/pages/Destinations/containers/CreateDestination/EmailSender/EmailSender.js

```diff
@@ -0,0 +1,136 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { EuiFlexGroup, EuiFlexItem, EuiButton } from '@elastic/eui';
+
+import { FormikComboBox } from '../../../../../components/FormControls';
+import { validateEmailSender } from './utils/validate';
+import { isInvalid, hasError } from '../../../../../utils/validate';
+import ManageSenders from '../ManageSenders';
+import getSenders from './utils/helpers';
+
+export default class EmailSender extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      senderOptions: [],
+      loadingSenders: true,
+      showManageSendersModal: false,
+    };
+
+    this.getSenders = getSenders.bind(this);
+    this.onClickManageSenders = this.onClickManageSenders.bind(this);
+    this.onClickCancel = this.onClickCancel.bind(this);
+    this.onClickSave = this.onClickSave.bind(this);
+  }
+
+  componentDidMount() {
+    this.loadSenders();
+  }
+
+  onClickManageSenders() {
+    this.setState({ showManageSendersModal: true });
+  }
+
+  onClickCancel() {
+    this.setState({ showManageSendersModal: false });
+  }
+
+  onClickSave = () => {
+    // TODO: Check if 'then' is necessary here
+    // this.setState({ showManageSendersModal: false })
+    this.loadSenders().then(r => this.setState({ showManageSendersModal: false }));
```

Comment:
What is behavior for this? onClickSave just loads senders and closes a modal?

Reply:
The `ManageSenders` and `ManageEmailGroups` modals are decoupled from their parent components since they're rendered in a couple places, however the parent component might still need to perform actions `onClose` and `onSave` so the bulk of the "processing form logic" is being done within the modals themselves when `Save` is clicked but any function passed in for `onClickSave` will also be run after that. So in this case after `Save` is clicked, `ManageSenders` will process the senders, then call this passed in function which will load the updated senders for the `EmailSender` component and then close the modal.